invenio_files_rest/views.py

```diff
@@ -205,56 +285,40 @@ def post(self, **kwargs):
             :statuscode 403: access denied
             :statuscode 500: failed request
         """
-        args = parser.parse(self.post_args, request)
-        try:
-            if args['location_name']:
-                # TODO: Check why query is used directly.
-                location = Location.get_by_name(args['location_name'])
-            else:
-                # Get one of the active locations
-                location = Location.get_default()
-            if not location:
-                abort(400, 'Invalid location.')
-            bucket = Bucket(
-                default_storage_class=current_app.config[
+        with db.session.begin_nested():
+            bucket = Bucket.create(
+                storage_class=current_app.config[
                     'FILES_REST_DEFAULT_STORAGE_CLASS'
                 ],
-                default_location=location.id
+                location=location_name
             )
-            db.session.add(bucket)
-            db.session.commit()
-        except SQLAlchemyError:
-            db.session.rollback()
-            current_app.logger.exception('Failed to create bucket.')
-            abort(500, 'Failed to create bucket.')
-
-        # TODO: Implement serializer
-        return {'json':
-                {'size': bucket.size,
-                 'url': url_for("invenio_files_rest.bucket_api",
-                                bucket_id=bucket.id, _external=True),
-                 'uuid': str(bucket.id)
-                 }
-                }
+        db.session.commit()
```

Comment:
Do we have error handlers registered, so that we don't get a 500 error if e.g. the user specifies an invalid location?